integrations/google_genai/src/haystack_integrations/components/embedders/google_genai/text_embedder.py

```diff
@@ -0,0 +1,136 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict, List, Optional
+
+from google import genai
+from google.genai import types
+from haystack import component, default_from_dict, default_to_dict, logging
+from haystack.utils import Secret, deserialize_secrets_inplace
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class GoogleAITextEmbedder:
+    """
+    Embeds strings using OpenAI models.
+
+    You can use it to embed user query and send it to an embedding Retriever.
+
+    ### Usage example
+
+    ```python
+    from haystack.components.embedders import GoogleAITextEmbedder
+
+    text_to_embed = "I love pizza!"
+
+    text_embedder = GoogleAITextEmbedder()
+
+    print(text_embedder.run(text_to_embed))
+
+    # {'embedding': [0.017020374536514282, -0.023255806416273117, ...],
+    # 'meta': {'model': 'text-embedding-004-v2',
+    #          'usage': {'prompt_tokens': 4, 'total_tokens': 4}}}
+    ```
+    """
+
+    def __init__(  # pylint: disable=too-many-positional-arguments
+        self,
+        api_key: Secret = Secret.from_env_var("GOOGLE_API_KEY"),
+        model: str = "text-embedding-004",
+        config: Optional[types.EmbedContentConfig] = types.EmbedContentConfig(
+            task_type="SEMANTIC_SIMILARITY"),
+
+    ):
+        """
+        Creates an GoogleAITextEmbedder component.
+
+        :param api_key:
+            The Google API key.
+            You can set it with an environment variable `GOOGLE_API_KEY`, or pass with this parameter
+            during initialization.
+        :param model:
+            The name of the model to use for calculating embeddings.
+            The default model is `text-embedding-004`.
```

Comment:
Update default name here too
```suggestion
            The default model is `text-embedding-005`.
```